agent/api/task.go

```diff
@@ -202,23 +208,64 @@ func (task *Task) initializeCredentialsEndpoint(credentialsManager credentials.M
 
 }
 
+// BuildCNIConfig constructs the cni configuration from eni
+func (task *Task) BuildCNIConfig() (*ecscni.Config, error) {
+	if !task.isNetworkModeVPC() {
+		return nil, errors.New("task isnot enabled with vpc networkmode")
+	}
+
+	cfg := &ecscni.Config{}
+	eni := task.GetTaskENI()
+	if eni == nil {
```

Comment:
Yes, but here it also needs the task eni information. Do you want me to change to:
```
if !task.isNetworkModeVPC(){ 
return errors.New()
}

eni := task.GetTaskENI()

```